python/lsst/summit/utils/astrometry/utils.py

```diff
@@ -0,0 +1,288 @@
+from astropy.coordinates import AltAz, EarthLocation, SkyCoord
+
+from lsst.afw.geom import SkyWcs
+from lsst.daf.base import PropertySet
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask, CharacterizeImageConfig
+
+__all__ = [
+    'chuckHeaderToWcs',
+    'getAverageRaFromHeader',
+    'getAverageDecFromHeader',
+    'getAverageAzFromHeader',
+    'getAverageElFromHeader',
+    'genericCameraHeaderToWcs',
+    'getIcrsAtZenith',
+    'headerToWcs',
+    'runCharactierizeImage',
+    'filterSourceCatOnBrightest',
+]
+
+
+def chuckHeaderToWcs(exp, nominalRa=None, nominalDec=None):
+    """Given an exposure taken by Chuck at his house, construct a wcs
+    with the ra/dec set to zenith unless a better guess is supplied.
+
+    Automatically sets the platescale depending on the lens.
+
+    Parameters
+    ----------
+    exp : `lsst.afw.image.Exposure`
+        The exposure to construct the wcs for.
+    nominalRa : `float`, optional
+        The guess for the ra.
+    nominalDec : `float`, optional
+        The guess for the Dec.
+
+    Returns
+    -------
+    wcs : `lsst.afw.geom.SkyWcs`
+        The constructed wcs.
+    """
+    header = exp.getMetadata().toDict()
+
+    # set the plate scale depending on the lens and put into CD matrix
+    # plate scale info from:
+    # https://confluence.lsstcorp.org/pages/viewpage.action?pageId=191987725
+    lens = header['INSTLEN']
+    if '135mm' in lens:
+        arcSecPerPix = 8.64
+    elif '375mm' in lens:
+        arcSecPerPix = 3.11
+    elif '750mm' in lens:
+        arcSecPerPix = 1.56
+    else:
+        raise ValueError(f'Unrecognised lens: {lens}')
+
+    header['CD1_1'] = arcSecPerPix / 3600
+    header['CD1_2'] = 0
+    header['CD2_1'] = 0
+    header['CD2_2'] = arcSecPerPix / 3600
+
+    # calculate the ra/dec at zenith and assume Chuck pointed it vertically
+    icrs = getIcrsAtZenith(float(header['OBSLON']),
+                           float(header['OBSLAT']),
+                           float(header['OBSHGT']),
+                           header['UTC'])
+    header['CRVAL1'] = nominalRa if nominalRa else icrs.ra.degree
+    header['CRVAL2'] = nominalDec if nominalDec else icrs.dec.degree
+
+    # just use the nomimal chip centre, not that it matters
+    # given radec = zenith
+    width, height = exp.image.array.shape
+    header['CRPIX1'] = width/2
+    header['CRPIX2'] = height/2
+
+    header['CTYPE1'] = 'RA---TAN-SIP'
+    header['CTYPE2'] = 'DEC--TAN-SIP'
+
+    wcsPropSet = PropertySet.from_mapping(header)
+    wcs = SkyWcs(wcsPropSet)
+    return wcs
+
+
+# don't be tempted to get cute and try to combine these 4 functions. It would
+# be a easy to do but it's not unlikley they will diverge in the future.
+def getAverageRaFromHeader(header):
+    raStart = header.get('RASTART')
+    raEnd = header.get('RAEND')
+    if not raStart or not raEnd:
+        raise RuntimeError(f'Failed to get RA from header due to missing RASTART/END {raStart} {raEnd}')
```

Comment:
Just FYI (because you probably shouldn't use it here where the all caps are meaningful), the `f{x=}` syntax is great.

Reply:
Thanks, I'll look those up!